api/v1alpha1/controllermanagerconfig_loader.go

```diff
@@ -45,12 +46,12 @@ func addOptionsFromConfigSpec(o *ctrl.Options, configSpec ControllerManagerConfi
 		o.Cache.SyncPeriod = &configSpec.SyncPeriod.Duration
 	}
 
-	if len(o.Cache.Namespaces) == 0 && configSpec.CacheNamespace != "" {
-		o.Cache.Namespaces = []string{configSpec.CacheNamespace}
+	if len(o.Cache.DefaultNamespaces) == 0 && configSpec.CacheNamespace != "" {
+		o.Cache.DefaultNamespaces[configSpec.CacheNamespace] = cache.Config{}
```

Comment:
controller runtime now supports to have a [map of allowed namespaces](https://github.com/kubernetes-sigs/controller-runtime/pull/2421),
to support that feature we must add new configurations  with the more granular configurations, 
may we add an issue to address that?